src/pages/callback/index.tsx

```diff
@@ -1,29 +1,28 @@
 import { useEffect } from 'react'
-import { useLocation, useNavigate } from 'react-router-dom'
+import { useNavigate, useSearchParams } from 'react-router-dom'
 import { fetchUserInfo } from '@/service'
 import Loading from '@/components/loading'
-import { tokenAtom, usernameAtom } from '@/store'
-import { useAtom } from 'jotai'
+import { isAdminAtom, store, tokenAtom, usernameAtom } from '@/store'
 
 const Callback = () => {
-  const location = useLocation()
+  const [searchParams] = useSearchParams()
   const navigate = useNavigate()
-  const [, setToken] = useAtom(tokenAtom)
-  const [, setUsername] = useAtom(usernameAtom)
 
   useEffect(() => {
-    const searchParams = new URLSearchParams(location.search)
     const code = searchParams.get('code')
     const redirectUri = searchParams.get('redirect_uri')
 
-    if (code) {
-      fetchUserInfo(code).then((data) => {
-        setToken(data.jwt)
-        setUsername(data.username)
-        navigate(redirectUri || '/')
-      })
+    const handleFetchUserInfo = async (code: string) => {
+      const userInfo = await fetchUserInfo(code)
+      store.set(tokenAtom, userInfo.jwt)
+      store.set(usernameAtom, userInfo.username)
+      store.set(isAdminAtom, userInfo.adminNamespaces.length > 0)
```

Comment:
what about admin projects?
In current version, just show admin-pull-request for adminNamespaces not null and show admin-task-apply for adminProjects not null.